tools/plugins/com.liferay.ide.server.core/src/com/liferay/ide/server/core/portal/PortalServer.java

```diff
@@ -12,19 +12,38 @@
  * details.
  *
  *******************************************************************************/
+
 package com.liferay.ide.server.core.portal;
 
+import com.liferay.ide.server.core.ILiferayServer;
 
 /**
  * @author Gregory Amerson
+ * @author Terry Jia
  */
-public interface PortalServer
+public interface PortalServer extends ILiferayServer
 {
 
     String START = "start";
     String STOP = "stop";
+
+    String PROPERTY_EXTERNAL_PROPERTIES = "externalProperties";
+
     String PROPERTY_MEMORY_ARGS = "memoryArgs";
-    
+
+    String PROPERTY_SERVER_MODE = "serverMode";
+
+    String PROPERTY_USE_DEFAULT_PORTAL_SERVER_SETTINGS = "useDefaultPortalServerSettings";
```

Comment:
We don't have to migrate old property names we can pick new ones since this is a new server type.  So for this one lets use:

PROPERTY_LAUNCH_SETTINGS